### ISSUE TITLE
Does the fsdp_sft_trainer support Qwen/Qwen3-30B-A3B?

### ISSUE THREAD

**Issue by XIRZC** [volcengine/verl#3380]
<img width="900" height="726" alt="Image" src="https://github.com/user-attachments/assets/48068685-bee9-4859-b92d-741f984de52a" />

<img width="938" height="770" alt="Image" src="https://github.com/user-attachments/assets/7410c76b-200b-4b1e-99fb-e9e2cf64a90a" />

**Comment by lanzhi44** [comment]
I encountered a similar problem when I use fsdp_sft_trainer for Qwen3-30B-A3B. If I don't enable the parameter `model.enable_gradient_checkpointing=False`, it causes torch to report an error: `A different number of tensors was saved during the original forward and recomputation.`; if I enable this parameter, then a timeout occurs similar to the situation described in this issue.

**Comment by LIYHUI** [comment]
have you solved it?

**Comment by Kirrito-k423** [comment]
中文：从当前主分支的公开证据看，`Qwen/Qwen3-30B-A3B` / `Qwen/Qwen3-30B-A3B-Base` 这类 MoE 模型的推荐 SFT 路径不是旧的 `fsdp_sft_trainer` 或 plain `engine=fsdp`。现在仓库里有 Qwen3-30B-A3B 的 SFT 示例，但它使用的是 `verl.trainer.sft_trainer` + `engine=automodel` + `engine.distributed_strategy=fsdp2` + `engine.ep_size=8`，并依赖 DeepEP / TransformerEngine / grouped_gemm 等 MoE/EP 相关能力。大 MoE 性能文档里，Qwen3-30B-A3B 的训练路径也主要指向 Megatron + EP。

你贴出的截图看起来是 `verl.trainer.fsdp_sft_trainer` 中的 NCCL collective/alltoall timeout；另一个用户也报告了同一模型在 gradient checkpointing 开关下分别出现 saved-tensor mismatch 或 timeout。基于目前 issue 中只有截图、没有完整启动命令和环境信息，我不建议把这理解为“plain FSDP SFT 已支持但只是配置问题”。更稳妥的尝试路径是：优先使用 `examples/sft/gsm8k/run_qwen3_30b_automodel.sh`，或者如果是 RL/大规模 MoE 训练则参考 Megatron 的 Qwen3-30B-A3B 示例。如果你仍然需要 plain FSDP 路径，请补充 verl commit、Torch/Transformers/CUDA/NCCL 版本、GPU/节点数、完整启动命令，以及 Automodel 示例是否能在同一环境复现。

English: Based on the current public evidence on `main`, the recommended SFT path for `Qwen/Qwen3-30B-A3B` / `Qwen/Qwen3-30B-A3B-Base` is not the old `fsdp_sft_trainer` or plain `engine=fsdp`. The repository now has a Qwen3-30B-A3B SFT example, but it uses `verl.trainer.sft_trainer` with `engine=automodel`, `engine.distributed_strategy=fsdp2`, and `engine.ep_size=8`, together with MoE/EP-related components such as DeepEP, TransformerEngine, and grouped_gemm. The large-MoE performance docs also point Qwen3-30B-A3B training mainly to the Megatron + EP path.

The screenshots in this issue look like an NCCL collective/alltoall timeout inside `verl.trainer.fsdp_sft_trainer`. Another user reported the same model hitting either a saved-tensor mismatch with gradient checkpointing or a timeout without it. Since the issue currently only includes screenshots and not the full launch command/environment, I would not treat this as confirmed plain-FSDP SFT support with only a configuration problem. The safer path is to try `examples/sft/gsm8k/run_qwen3_30b_automodel.sh`, or use the Megatron Qwen3-30B-A3B example for large-scale MoE/RL training. If you specifically need the plain FSDP path, please share the verl commit, Torch/Transformers/CUDA/NCCL versions, GPU/node topology, full launch command, and whether the Automodel example reproduces in the same environment.
This response was generated by a scheduled Codex GPT5.5 issue processing task.
